scripts/kata-install/osc-kata-addons-install.sh

```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
+#
+# osc-kata-addons-install.sh
+# Install addon artifacts (kernel/initrd) from container images
+# Reuses functions from lib.sh
+#
+
+set -e
+
+# Source the shared library functions
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+source "${SCRIPT_DIR}/lib.sh"
+
+#######################################
+# Update provider-specific configuration file
+# Arguments:
+#   $1: kernel path (empty if not installed)
+# Returns:
+#   0 on success
+#######################################
+update_provider_config() {
+    local kernel_path="$1"
+    
+    local config_file="/etc/kata-containers/kata-se/configuration.toml"
+    
+    if [ ! -f  chroot /host "$config_file" ]; then
+        echo "Warning: Configuration file not found: $config_file"
+        return 1
+    fi
+    
+    echo "Updating configuration: $config_file"
+    
+    # Backup config
+    chroot /host cp "$config_file" "${config_file}.backup-$(date +%s)"
+    
+    # Update kernel if provided
+    if [ -n "$kernel_path" ]; then
+        chroot /host sed -i "s|^\(kernel[[:space:]]*=[[:space:]]*\)\".*\"|\1\"$kernel_path\"|g" "$config_file"
+        echo "  Updated kernel: $kernel_path"
+    fi
+
+    return 0
+}
+
+#######################################
+# Install addon artifacts
+# Reads from environment variables
+# Returns:
+#   0 on success
+#######################################
+install_addons() {
+    local addon_image="${ADDON_IMAGE:-}"
+    
+    if [ -z "$addon_image" ]; then
+        echo "No addon image configured"
+        return 0
+    fi
+    
+    echo "Installing addon artifacts from: $addon_image"
+    
+    local kernel_src="${ADDON_KERNEL_PATH:-}"
```

Comment:
It looks like a good place to bail out if ADDON_KERNEL_PATH is no kernel path is provided.